saleor/payment/utils.py

```diff
@@ -433,3 +433,29 @@ def price_to_minor_unit(value: Decimal, currency: str):
     number_places = Decimal("10.0") ** precision
     value_without_comma = value * number_places
     return str(value_without_comma.quantize(Decimal("1")))
+
+
+def get_channel_slug_from_payment(payment: Payment) -> Optional[str]:
+    channel_slug = None
+
+    if payment.checkout:
+        channel_slug = payment.checkout.channel.slug
+    elif payment.order:
+        channel_slug = payment.order.channel.slug
+
+    return channel_slug
+
+
+def try_void_or_refund_inactive_payment(
+    payment: Payment, transaction: Transaction, manager: "PluginsManager"
+):
+    """Handle refund or void inactive payments.
+
+    In case when we have open multiple payments for single checkout but only one is
+    active. Some payment methods don't required confirmation so we can receive delayed
+    webhook when we have order already paid.
+    """
+    if transaction.is_success:
+        update_payment_charge_status(payment, transaction)
+        channel_slug = get_channel_slug_from_payment(payment)
+        gateway.payment_refund_or_void(payment, manager, channel_slug=channel_slug)
```

Comment:
This can raise a PaymentError which will result in the webhook failing. On the partial payments branch we have a task called `refund_or_void_inactive_payment` that retries on PaymentError

Reply:
I tested this by running this code by our task. I created an iDEAL payment but didn't press the button to redirect back to Saleor and then changed my mind and paid with a VISA card. At this moment Saleor was only aware of the VISA transaction:
<img width="1433" alt="image" src="https://user-images.githubusercontent.com/2566928/142679221-5f1463cb-e41d-43c0-8d75-900977e9bb9f.png">

Later on, after the payment got captured, it got refunded both in Saleor and Adyen (and appeared on the list of payments):
<img width="869" alt="image" src="https://user-images.githubusercontent.com/2566928/142679951-4d448b57-5420-4dd3-a1a9-4bb925ccf6ff.png">

<img width="1177" alt="image" src="https://user-images.githubusercontent.com/2566928/142680232-0e429f44-0ee7-4b27-bcaa-8d67bf042fc3.png">